doc/source/whatsnew/v1.1.0.rst

```diff
@@ -1107,6 +1107,7 @@ Plotting
 - Bug in :meth:`DataFrame.plot.scatter` was adding a colorbar to the plot even if the argument ``c`` was assigned to a column containing color names (:issue:`34316`)
 - Bug in :meth:`pandas.plotting.bootstrap_plot` was causing cluttered axes and overlapping labels (:issue:`34905`)
 - Bug in :meth:`DataFrame.plot.scatter` caused an error when plotting variable marker sizes (:issue:`32904`)
+- Twinned axes were losing their tick labels which should only happen to all but the last row or column of 'externally' shared axes (:issue:`33767`)
```

Comment:
a small note: this will need to move to 1.2 now ^^

Reply:
Yup :) I'm not managing to add a commit here, not sure why, I'm getting
```
error: src refspec ebardie/fix_twinx_sharex_tickx_label_confusion does not match any
error: failed to push some refs to 'git@github.com:ebardie/pandas.git'
```
 - @ebardie do you want to move the whatsnew note to v1.2.0?

@charlesdong1991 apart from that, do think it's good to go?

Reply:
@MarcoGorelli  did you `git commit -m 'some text'` before push? i saw this error before when I forgot to add commit text.

I think we should hold up a bit, this error seems related to this change.
If you have fixed the commit here, could you try below to replace the first part in test in `test_has_externally_shared_axis`:

```python
fig = self.plt.figure()
plots = fig.subplots(2, 4)

# Create *externally* shared axes for first and third columns
plots[0][0] = fig.add_subplot(231, sharex=plots[1][0])
plots[0][2] = fig.add_subplot(233, sharex=plots[1][2])
```

Reply:
> a small note: this will need to move to 1.2 now ^^

Thanks. I've done this.